tests/integration.rs

```diff
@@ -604,3 +604,67 @@ fn no_git_folder_after_decompression_with_gitignore_flag_active() {
         ".git folder should not exist after decompression"
     );
 }
+
+#[cfg(feature = "allow_piped_choice")]
+#[proptest(cases = 25)]
+fn unpack_multiple_sources_into_the_same_destination_with_merge(
+    ext: DirectoryExtension,
+    #[any(size_range(0..1).lift())] extra_extensions: Vec<FileExtension>,
+) {
+    let temp_dir = tempdir()?;
+    let root_path = temp_dir.path();
+    let source_path = root_path
+        .join(format!("example_{}", merge_extensions(&ext, &extra_extensions)))
+        .join("sub_a")
+        .join("sub_b")
+        .join("sub_c");
+
+    fs::create_dir_all(&source_path)?;
+    let archive = root_path.join(format!("archive.{}", merge_extensions(&ext, &extra_extensions)));
+    crate::utils::cargo_bin()
+        .arg("compress")
+        .args([
+            fs::File::create(source_path.join("file1.txt"))?.path(),
+            fs::File::create(source_path.join("file2.txt"))?.path(),
+            fs::File::create(source_path.join("file3.txt"))?.path(),
+        ])
+        .arg(&archive)
+        .assert()
+        .success();
+
+    fs::remove_dir_all(&source_path)?;
+    fs::create_dir_all(&source_path)?;
+    let archive1 = root_path.join(format!("archive1.{}", merge_extensions(&ext, &extra_extensions)));
+    crate::utils::cargo_bin()
+        .arg("compress")
+        .args([
+            fs::File::create(source_path.join("file4.txt"))?.path(),
+            fs::File::create(source_path.join("file5.txt"))?.path(),
+            fs::File::create(source_path.join("file6.txt"))?.path(),
```

Comment:
What happens if you change the two archives to be:

`1 2 3`
and 
`3 4 5` (don't have the 6th)

Does it ask to overwrite again?

Reply:
I updated the test case, and it does not ask for overwriting again.

Reply:
I don't think this is ideal, after telling Ouch to merge folders, I would expect it to prompt me again about conflicts, or, we can have folder merging enabled by default and then receive the usual prompts for conflicts asking if I want to overwrite or not.

For reference, the latter is how the 7z GUI works.

Reply:
hmmm, yes, it is possible, but it needs to refactor the question.rs file very hard.
you can put me in another PR to do that.
WDYT.

Reply:
Cool :) 

I'd love to see another PR for that but don't feel too pressured, I'd understand if you're not able to put the time